### DISCUSSION TITLE
getValues() expected value when using FormProvider?

### DISCUSSION THREAD

**Discussion by changyeamoon** [react-hook-form/react-hook-form#2907] [Q&A]
CSB - https://codesandbox.io/s/react-hook-form-wizard-form-provider-context-3snoj?file=/src/index.js:727-731

Hi, I have a `FormProvider` wrapping a multistep form. So I use `useFormContext` to get the methods I need instead of wrapping each step with a form from `useForm`. 

To me, the expected value from `getValues()` at any step should be the entire form object or `defaultValues` with the new field changes.

In the original WizardForm Example - https://codesandbox.io/s/react-hook-form-wizard-form-gxvvc , It makes sense that each step would only output the data for that particular step, since theyre individually wrapped in a form. But it my case I only have one form and `FormProvider` wrapping all the steps. Why am I wrong to assume that `getValues()` should return the entire form object?

**Comment by bluebill1049** [comment]
getValues return the current form values, you want the entire funnel, you should read from your store and merge with what you have in the form right now.

**Reply by changyeamoon** [reply]
am i wrong to assume that the default values are the current form values?

in the beggining at least

**Reply by bluebill1049** [reply]
it's not. it's just defaultValues for the form.

<input defaultValue="test" />

so `getValues()` will return `""`

but you can always host your `defaultValues` to a variable.
